enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,500 @@
+versions, then upgrading/downgrading could change whether that resource is excluded by
+the CVO or not.  Once created, the CVO normally never deletes resources, so some manual
+cleanup might be needed to achieve the desired state.  For downgrades this is probably
+acceptable, for upgrades this could be a concern (resource A wasn't excluded in v1, but is
+excluded in v2.  Clusters that upgrade from v1 to v2 will still have resource A, but clusters
+installed at v2 will not have it).  Technically this situation can already arise today
+if a resource is deleted from the payload between versions.  Since the CVO can actually
+delete properly annotated resources, there may be an option here to make the CVO automatically
+delete "now filtered" resources, but that brings us back to the aforementioned issues
+with ensuring total cleanup of a filtered component.  Manual cleanup/release notes may be
+the most comprehensive solution.
+
+
+
+### Version Skew Strategy
+
+N/A
+
+### Operational Aspects of API Extensions
+
+N/A - no api extensions are being introduced in this proposal
+
+#### Failure Modes
+
+N/A - no api extensions are being introduced in this proposal
+
+#### Support Procedures
+
+N/A - no api extensions are being introduced in this proposal
+
+## Implementation History
+
+N/A
+
+## Drawbacks
+
+The primary drawback is that this increases the matrix of cluster configurations/topologies and
+the behavior that is expected from each permutation.
+
+## Alternatives
+
+* CVO already supports a CLUSTER_PROFILE env variable.  We could define specific profiles like "headless"
+that disables the console.  CLUSTER_PROFILE isn't a great fit because the idea there is to define a relatively
+small set of profiles to define specific sets of components to be included, not to allow a user to fully pick
+and choose individual components.  We would have to define a large set of profiles to encompass all the possible
+combinations of components to be enabled/disabled.  Direction has been set that cluster profiles should only be
+used defining the standalone and hypershift form factors.
+
+* CVO already supports an EXCLUDE_MANIFESTS env variable which is used to implement the ROKS deployment topology.
+Unfortunately it only allows a single annotation to be specified, so even if we want to use it for this purpose
+it needs to be extended to support multiple annotations so multiple individual components can be excluded
+independently rather than requiring all components to be excluded to share a single common annotation.
+
+Regardless we need a way to expose this configuration as a first class part of the install config provided by the
+user creating the cluster, so at a minimum we need to add a mechanism to wire an install config value into
+the CVO arguments and allow the CVO to consume more than a single annotation to exclude.
+
+* Allow the installer to specify additional resources to `include` in addition to ones to `exclude`.  This has the challenge
```

Comment:
this alternative has since been accepted, and the safety valve, as discussed above^^, is that if a capability is in `include` but has no manifests matching the cluster-profile, you don't get any new manifests added.  I.e. you have successfully added that profile's empty in-cluster implementation of that capability.  Unclear to me if folks running HyperShift clusters will want the knob to be able to opt in to or out of any host-cluster-side components.  Looking at your example capability list, and thinking over my fuzzy understanding of which components are host-cluster-side in HyperShift, it seems like all the currently floated capabilities are guest-cluster-side anyway?

Reply:
> Looking at your example capability list, and thinking over my fuzzy understanding of which components are host-cluster-side in HyperShift, it seems like all the currently floated capabilities are guest-cluster-side anyway?

mostly.  Things that run on the host side tend to be non-optional (control plane stuff).  The items that might be borderline are:

1) OLM (runs on the host, but maybe someone doesn't want to be able to extend their cluster w/ olm, someday.  unclear if we'd ever make it truly optional since we have plans to leverage it for more core functionality, though)
2) monitoring.  I believe the prom stack runs on the host side today(though i'm not positive on that), it's possible (if unlikely) someone looking for a very lightweight cluster might want to disable monitoring in the future.

if either of those things happen, the hypershift installer/host-controller would have to implement similar discriminating functionality.  Not a concern in this pass.